src/main/java/org/threadly/litesockets/utils/IOUtils.java

```diff
@@ -182,18 +193,21 @@ public int read(byte[] ba, int offset, int len) throws IOException {
     
     @Override
     public int read() throws IOException {
+      if(isClosed) {
+        return -1;
+      }
       while(true) {
-        synchronized(this) {
+        synchronized(currentBB) {
```

Comment:
It seems like you should synchronized before the while loop to prevent releasing and then immediately re-acquiring the lock as you loop